src/homepageExperience/components/steps/arduino/InitializeClient.tsx

```diff
@@ -86,14 +86,14 @@ export const InitializeClient: FC<OwnProps> = ({
       dispatch(createAuthorization(authorization))
       event(`firstMile.arduinoWizard.tokens.tokenCreated`)
     }
-  }, [sortedPermissionTypes.length])
+  }, [dispatch, me.id, org.id, sortedPermissionTypes, tokenValue])
```

Comment:
do we need to use optional chaining (`?.`) for `me.id` and `org.id`?

Reply:
I don't believe so, I think we confirm these are non-null earlier on.

Reply:
I don't see confirmation these aren't `null` or `undefined` earlier, so I'd add the optional chaining. There's two things to think about here. 

1. If this is coming from state (e.g., Redux, Context), what's the default value, and is it the type we expect?

For example, in `/me/reducers`, we can see `me.id` should be a `string` and exist.  Great.

2. However, even if it's coming from state, and there's a default state, is it going to get reassigned after the user data is retrieved to something we don't expect? 

In `/me/reducers`, `me.id` gets set (action name is `SET_ME`) to whatever the backend sends us for that property - there isn't any type validation logic in the reducer. So while IDPE _shouldn't_ be sending us a null or undefined id, there's nothing on our side of the house to prevent that from happening.